services/api/src/resources/file/resolvers.ts

```diff
@@ -46,6 +46,23 @@ export const getDownloadLink: ResolverFn = async ({ s3Key }, input, { userActivi
   });
 };
 
+export const getDownloadLinkByTaskFileId: ResolverFn = async (
+  root,
+  { taskId, fileId },
+  { sqlClientPool, hasPermission, userActivityLogger }
+) => {
+  const rowsPerms = await query(sqlClientPool, taskSql.selectPermsForTask(taskId));
+
+  await hasPermission('task', 'view', {
+    project: R.path(['0', 'pid'], rowsPerms)
+  });
+
+  const rows = await query(sqlClientPool, Sql.selectTaskFileById(taskId, fileId));
+
+  const downloadUrl = await getDownloadLink(rows[0], {}, userActivityLogger)
```

Comment:
The third argument to `getDownloadLink` should be a [`context`](https://github.com/uselagoon/lagoon/blob/main/services/api/src/resources/index.ts#L14) but this is only passing `userActivityLogger`. The result is that no activity/audit logs are made when invoking `getDownloadLinkByTaskFileId`.